### PR TITLE
implementing get/set performedCode in the contractDetails

### PR DESCRIPTION
**Pull request by AionJayT**
## Notice

It is not allowed to submit your PR to the master branch directly, please submit your PR to the master-pre-merge branch.

## Description
- The avm requires to store two different code (deployed code for consensus and transformed code for executing) in the kernel repo. This PR implemented new interfaces also created another database for storing the avm executing code. 
- The changes in AVM is here: https://github.com/aionnetwork/AVM/pull/387

Fixes Issue # N/A

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- 

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [ ] My code generates no new warnings.
- [ ] Any dependent changes have been made.

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java [hunk 1]
```diff
@@ -309,6 +314,15 @@ public void saveCode(Address address, byte[] code) {
         return getContractDetails(address).getCode(codeHash);
     }
 
+    @Override
+    public byte[] getTransformedCode(Address address) {
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: This method requires locking. I suggest using the full write locks to ensure correctness. We can improve performance later.

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java [hunk 2]
```diff
@@ -624,4 +638,25 @@ public boolean isIndexed(byte[] hash, long level) {
     public byte getVMUsed(Address contract) {
         return repository.getVMUsed(contract);
     }
+
+    @Override
+    public void setTransformedCode(Address contractAddr, byte[] code) {
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: Requires write locks.

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java (hunk 1 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: for the getTransformedCode probably don't need locks? the Transformed code shouldn't be changed after deploy. Like the getcode method, they don't have locks also.

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java (hunk 2 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: sure. will add it.

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java (hunk 1 above)
Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: to get the code we often need to load the accounts which requires write locks

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java (hunk 2 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java]: https://github.com/aionnetwork/aion/pull/870/commits/cd6b10d512065c88293c4befad3dcb3d0fe56307

**Pull request merged**